internal/handlers/handlers.go

```diff
@@ -0,0 +1,88 @@
+package handlers
+
+import (
+	"context"
+	// nolint:gosec // pprof path is only exposed over localhost
+	_ "net/http/pprof"
+
+	"github.com/bmc-toolbox/bmclib/v2"
+	"github.com/metal-toolbox/bioscfg/internal/log"
+	"github.com/metal-toolbox/bioscfg/internal/model"
+	"github.com/metal-toolbox/bioscfg/internal/store"
+	"github.com/metal-toolbox/bioscfg/internal/tasks"
+	rctypes "github.com/metal-toolbox/rivets/condition"
+	"github.com/metal-toolbox/rivets/events/controller"
+	"github.com/pkg/errors"
+)
+
+// HandlerFactory has the data and business logic for the application
+type HandlerFactory struct {
+	repository store.Repository
+}
+
+// NewHandlerFactory returns a new instance of the Handler
+func NewHandlerFactory(repository store.Repository) *HandlerFactory {
+	return &HandlerFactory{
+		repository: repository,
+	}
+}
+
+func (h *HandlerFactory) parseParams(condition *rctypes.Condition) (*rctypes.BiosControlTaskParameters, error) {
+	params, err := rctypes.NewBiosControlParametersFromCondition(condition)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to parse condition parameters")
+	}
+
+	log.WithField("params", params).Debug("Parsed condition parameters")
+
+	return params, nil
+}
+
+func (h *HandlerFactory) getAsset(ctx context.Context, params *rctypes.BiosControlTaskParameters) (*model.Asset, error) {
+	asset, err := h.repository.AssetByID(ctx, params.AssetID)
+	if err != nil {
+		// TODO: Check error type
+		return nil, controller.ErrRetryHandler
+	}
+
+	log.With(asset).Debug("Found asset")
+
+	return asset, nil
+}
+
+func (h *HandlerFactory) Handle(
+	ctx context.Context,
+	condition *rctypes.Condition,
+	publisher controller.ConditionStatusPublisher,
+) error {
+	log.WithField("condition", condition).Debug("Handling condition")
+
+	params, err := h.parseParams(condition)
+	if err != nil {
+		return err
+	}
+
+	asset, err := h.getAsset(ctx, params)
+	if err != nil {
+		return err
+	}
+
+	var task tasks.Task
+
+	switch params.Action {
+	case rctypes.ResetSettings:
+		task = tasks.NewBiosResetTask(asset)
+	default:
+		log.With(asset).WithField("action", params.Action).Error("Invalid action")
+		return model.ErrInvalidAction
+	}
+
+	runner := tasks.NewTaskRunner(publisher, task)
+	client := bmclib.NewClient(asset.BmcAddress.String(), asset.BmcUsername, asset.BmcPassword)
+
+	if err := runner.Run(ctx, client); err != nil {
```

Comment:
does the runner catch panics?
for an example see https://github.com/metal-toolbox/flasher/blob/56b74a06018ba46b38bd82ccd48eaf531f7571fc/internal/runner/runner.go#L70

Reply:
I left a TODO in there before, and forgot to get back to it, but handling it now.